crates/qt-build-utils/src/lib.rs

```diff
@@ -1030,9 +1031,12 @@ Q_IMPORT_PLUGIN({plugin_class_name});
 
     /// Run [rcc](https://doc.qt.io/qt-6/resources.html) on a .qrc file and save the output into [cargo's OUT_DIR](https://doc.rust-lang.org/cargo/reference/environment-variables.html).
     /// The path to the generated C++ file is returned, which can then be passed to [cc::Build::files](https://docs.rs/cc/latest/cc/struct.Build.html#method.file).
-    /// The compiled static library must be linked with [+whole-archive](https://doc.rust-lang.org/rustc/command-line-arguments.html#linking-modifiers-whole-archive)
-    /// or the linker will discard the generated static variables because they are not referenced from `main`.
-    pub fn qrc(&mut self, input_file: &impl AsRef<Path>) -> PathBuf {
+    /// This function also returns a String that contains the name of the resource initializer
+    /// function.
+    /// The build system must ensure that if the .cpp file is built into a static library, either
+    /// the `+whole-archive` flag is used, or the initializer function is called by the
+    /// application.
+    pub fn qrc(&mut self, input_file: &impl AsRef<Path>) -> (PathBuf, String) {
```

Comment:
wonder if in the refactor we should have a type for a ProcessedFile or something that then has input path, source, output path, any initialiser function etc wrapped up.